From 1643c49ed378df5b67c365b0a9e039c1e67c023a Mon Sep 17 00:00:00 2001
From: Yussuf Shaikh <yussuf@us.ibm.com>
Date: Thu, 27 Apr 2017 13:41:22 -0700
Subject: [PATCH 1086/1164] CDH-53202 HIVE-16538 : TestExecDriver fails if run
 after TestOperators#testScriptOperator (Yussuf
 Shaikh via Ashutosh Chauhan)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
(cherry picked from commit 2f5a2a79d1bf1f8b418c731be7040dea0d85a39a)

Change-Id: I9971ada4960a75641eecce258ca0d64201475a33
---
 .../apache/hadoop/hive/ql/exec/TestOperators.java  |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/ql/src/test/org/apache/hadoop/hive/ql/exec/TestOperators.java b/ql/src/test/org/apache/hadoop/hive/ql/exec/TestOperators.java
index 0049f53..f661528 100644
--- a/ql/src/test/org/apache/hadoop/hive/ql/exec/TestOperators.java
+++ b/ql/src/test/org/apache/hadoop/hive/ql/exec/TestOperators.java
@@ -35,6 +35,7 @@
 import org.apache.hadoop.hive.ql.io.IOContext;
 import org.apache.hadoop.hive.ql.parse.TypeCheckProcFactory;
 import org.apache.hadoop.hive.ql.plan.CollectDesc;
+import org.apache.hadoop.hive.ql.plan.ExprNodeColumnDesc;
 import org.apache.hadoop.hive.ql.plan.ExprNodeConstantDesc;
 import org.apache.hadoop.hive.ql.plan.ExprNodeDesc;
 import org.apache.hadoop.hive.ql.plan.MapredWork;
@@ -53,6 +54,7 @@
 import org.apache.hadoop.hive.serde2.objectinspector.StructField;
 import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
 import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapred.InputSplit;
 import org.apache.hadoop.mapred.JobConf;
@@ -195,10 +197,11 @@ public void testScriptOperator() throws Throwable {
     try {
       System.out.println("Testing Script Operator");
       // col1
-      ExprNodeDesc exprDesc1 = TestExecDriver.getStringColumn("col1");
-
+      ExprNodeDesc exprDesc1 = new ExprNodeColumnDesc(TypeInfoFactory.stringTypeInfo, "col1", "",
+          false);
       // col2
-      ExprNodeDesc expr1 = TestExecDriver.getStringColumn("col0");
+      ExprNodeDesc expr1 = new ExprNodeColumnDesc(TypeInfoFactory.stringTypeInfo, "col0", "",
+          false);
       ExprNodeDesc expr2 = new ExprNodeConstantDesc("1");
       ExprNodeDesc exprDesc2 = TypeCheckProcFactory.DefaultExprProcessor
           .getFuncExprNodeDesc("concat", expr1, expr2);
-- 
1.7.9.5

